### PR TITLE
PLAT-84230: Cleanup interfaces, code, and JS Doc for VirtualList with different item size

### PR DESCRIPTION
**Pull request by ybsung**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [X] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

To deliver VirtualList with different item size, it is needed to clean up interface, code, and JS doc for it.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

Cleanup interfaces, code, and JS Doc for VirtualList with different item size

### Links
[//]: # (Related issues, references)

PLAT-84230

### CODE REVIEW

File: packages/moonstone/VirtualList/VirtualList.js [hunk 1]
```diff
@@ -41,16 +42,27 @@ const VirtualList = kind({
 		 * <VirtualList itemSize={ri.scale(72)} />
 		 * ```
 		 *
-		 * @type {Number}
+		 * @type {Number||ui/VirtualList.listDifferentItemSizeShape}
```
Comment (0x64) [inline packages/moonstone/VirtualList/VirtualList.js]: Double pipe character.

Comment (ybsung) [reply packages/moonstone/VirtualList/VirtualList.js]: Fixed it.

File: packages/ui/VirtualList/VirtualListBase.js [hunk 2]
```diff
@@ -21,15 +21,30 @@ const
...
+ *
+ * @typedef {Object} listDifferentItemSizeShape
+ * @memberof ui/VirtualList
+ * @property {Number} minWidth The minimum width of the grid list item.
+ * @property {Number[]} minHeight The minimum height of the grid list item.
+ * @public
+ */
+const listDifferentItemSizeShape = PropTypes.shape({
+	minSize: PropTypes.number.isRequired,
+	size: PropTypes.arrayOf(PropTypes.number).isRequired
```
Comment (0x64) [inline packages/ui/VirtualList/VirtualListBase.js]: Could you please check this? It probably is not a 'required' one.

Comment (ybsung) [reply packages/ui/VirtualList/VirtualListBase.js]: Good catch! When any item size is not known, then app developers could skip to pass `size` property in `itemSize` prop.

**Pull request merged**